src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -121,6 +118,10 @@ public Task<AuthenticationScheme> GetDefaultSignOutSchemeAsync()
             {
                 return GetSchemeAsync(_options.DefaultSignOutScheme);
             }
+            if (_signOutHandlers.Count() == 1)
+            {
+                return Task.FromResult(_signOutHandlers[0]);
+            }
             return GetDefaultSignInSchemeAsync();
```

Comment:
This problem also exists for the fallback for SignIn => Authenticate, so we need to add a similar check for that too